src/main/java/org/springframework/data/elasticsearch/client/reactive/DefaultReactiveElasticsearchClient.java

```diff
@@ -0,0 +1,518 @@
+		return this.hostProvider.getActive(VerificationMode.LAZY) //
+				.flatMap(it -> callback.doWithClient(it)) //
+				.onErrorResume(throwable -> {
+
+					if (throwable instanceof ConnectException) {
+
+						return hostProvider.getActive(VerificationMode.FORCE) //
+								.flatMap(webClient -> callback.doWithClient(webClient));
+					}
+
+					return Mono.error(throwable);
+				});
+	}
+
+	@Override
+	public Mono<Status> status() {
+
+		return hostProvider.clusterInfo() //
+				.map(it -> new ClientStatus(it.getNodes()));
+	}
+
+	// --> Private Response helpers
+
+	private static GetResult getResponseToGetResult(GetResponse response) {
+
+		return new GetResult(response.getIndex(), response.getType(), response.getId(), response.getVersion(),
+				response.isExists(), response.getSourceAsBytesRef(), response.getFields());
+	}
+
+	// -->
+
+	private <Req extends ActionRequest, Resp extends ActionResponse> Flux<Resp> sendRequest(Req request,
+			Function<Req, Request> converter, Class<Resp> responseType, HttpHeaders headers) {
+		return sendRequest(converter.apply(request), responseType, headers);
+	}
+
+	private <AR extends ActionResponse> Flux<AR> sendRequest(Request request, Class<AR> responseType,
+			HttpHeaders headers) {
+
+		return execute(webClient -> sendRequest(webClient, request, headers))
+				.flatMapMany(response -> readResponseBody(request, response, responseType));
+	}
+
+	private Mono<ClientResponse> sendRequest(WebClient webClient, Request request, HttpHeaders headers) {
+
+		RequestBodySpec requestBodySpec = webClient.method(HttpMethod.valueOf(request.getMethod().toUpperCase())) //
+				.uri(request.getEndpoint(), request.getParameters()) //
+				.headers(theHeaders -> theHeaders.addAll(headers));
+
+		if (request.getEntity() != null) {
+
+			requestBodySpec.contentType(MediaType.valueOf(request.getEntity().getContentType().getValue()));
+			requestBodySpec.body(bodyExtractor(request), String.class);
+		}
+
+		return requestBodySpec //
+				.exchange() //
+				.onErrorReturn(ConnectException.class, ClientResponse.create(HttpStatus.SERVICE_UNAVAILABLE).build());
+	}
+
+	private Publisher<String> bodyExtractor(Request request) {
+
+		return Mono.fromSupplier(() -> {
+
+			try {
+				return EntityUtils.toString(request.getEntity());
+			} catch (IOException e) {
+				throw new RequestBodyEncodingException("Error encoding request", e);
+			}
+		});
+	}
+
+	private <T> Publisher<? extends T> readResponseBody(Request request, ClientResponse response, Class<T> responseType) {
+
+		if (RawActionResponse.class.equals(responseType)) {
+			return Mono.just((T) new RawActionResponse(response));
+		}
+
+		if (response.statusCode().is5xxServerError()) {
+
+			throw new HttpClientErrorException(response.statusCode(),
+					String.format("%s request to %s returned error code %s.", request.getMethod(), request.getEndpoint(),
+							response.statusCode().value()));
+		}
+
+		return response.body(BodyExtractors.toDataBuffers()).flatMap(it -> {
```

Comment:
`toDataBuffers` streams response as chunks resulting in multiple contents. Shouldn't this rather await full response completion before decode bytes to a single `String`?